### PR TITLE
[cuda] fix: hide nvml-only unstartable gpu listings

### PR DESCRIPTION
**Pull request by Wangmerlyn**
## Summary
- Gate CUDA NVML list_gpus records on Torch CUDA availability, matching visible count, and successful `set_device()` probes for each visible ordinal.
- Fall back from untrusted NVML to Torch CUDA, and avoid non-HIP ROCm SMI fallback that could mislabel CUDA ordinals.
- Update AGENTS, README, architecture, CLI/MCP/Python/API docs, and add a plan with RED/GREEN evidence.

## Test Plan
- [x] `PYTHONPATH=$PWD/src pytest tests/utilities/test_gpu_info.py -q`
- [x] `PYTHONPATH=$PWD/src pytest tests/mcp tests/utilities/test_gpu_info.py -q`
- [x] `PYTHONPATH=$PWD/src pytest tests/global_controller tests/utilities/test_gpu_info.py -q`
- [x] `PYTHONPATH=$PWD/src pytest tests -q`
- [x] `PYTHONPATH=$PWD/src mkdocs build`
- [x] `pre-commit run --all-files`
- [x] `git diff --check`

## Local Review
- [x] CUDA/startability review found fallback and set_device gaps; fixed with RED regressions.
- [x] Tests/docs/hygiene review found set_device gap and architecture doc gap; fixed.
- [x] Final follow-up review: ready to merge, no remaining issues.

## Summary by CodeRabbit

* **Bug Fixes**
  * GPU listings now only show CUDA devices when they can actually be started by the runtime, preventing unusable entries from appearing.
  * Hidden NVML-only devices are no longer advertised as selectable GPU targets.

* **Documentation**
  * Updated CLI, API, architecture, Python, and reference docs to clarify CUDA/NVML visibility rules and startup behavior.
  * Added a new planning note describing the aligned GPU listing approach.

* **Tests**
  * Expanded GPU visibility test coverage for CUDA availability, masking, fallback, and startup edge cases.

### CODE REVIEW

File: docs/plans/cuda-nvml-startable-listing.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+## Tasks
+
+- [x] Add RED tests for Torch CUDA unavailable, zero count, device-count failure,
+      and CUDA visibility/Torch count mismatch.
+- [x] Implement the minimal NVML guard and Torch fallback behavior.
+- [x] Update `AGENTS.md`, README, CLI reference, MCP guide, and Python/API docs.
+- [x] Run focused GPU info tests.
+- [x] Run targeted GPU/MCP and global-controller-adjacent tests.
+- [x] Run full tests, docs build, hooks, and whitespace checks.
+- [ ] Request local subagent code review and resolve findings before PR.
```
[the remark this answers is not in the document]
Comment (Wangmerlyn) [reply docs/plans/cuda-nvml-startable-listing.md]: Fixed in 8c6165f by marking the local subagent code review checklist item complete so the plan status matches the verification log. Re-ran pre-commit, mkdocs build, and git diff --check for the doc-only follow-up.
